msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/BiGaussian.java

```diff
@@ -0,0 +1,185 @@
+		BiGaussianParams.sigmaRight = ((interpolationRightSideX - mu))/Math.sqrt(2*Math.log(2));
+		
+	}
+	
+	
+	/**
+	 * <p>
+	 * InterpolationX is used to calculate X value of Halfwidth-halfheight point of either left or right half of BiGaussian.
+	 * </p>
+	 * 
+	 * @param mu a {@link java.lang.Integer} object. This is X value for maximum height in terms of scan number.
+	 * @param halfHeight  a {@link java.lang.Double} object. This is m/z value from the raw file.
+	 * @param leftBound a {@link java.lang.Integer} object. This is minimum scan number.
+	 * @param rightBound a {@link java.lang.Integer} object. This is maximum scan number.
+	 * @param roundedmz a {@link java.lang.Integer} object. This is rounded m/z value.
+	 * @param direction a {@link Enum} object. This decides for which half we want to calculate X value.
+	 */
+	private double InterpolationX(int mu,double halfHeight,int leftBound,int rightBound,int roundedmz,Direction direction){
+		
+		int i = mu;
+		double  interpolationY1 = 0;
+		double interpolationY2 = 0;
+		int previousX2 = direction == Direction.RIGHT ? 1 : -1;
+		
+		while(leftBound <= i && i <= rightBound) {
+			
+			i += previousX2;
+			SliceSparseMatrix.Triplet triplet1 = horizontalSlice.get(i,roundedmz);
+			
+			if(triplet1!=null){
+				interpolationY1 = triplet1.intensity;
+			}
+			
+			if(interpolationY1 <halfHeight && triplet1!=null){
+				SliceSparseMatrix.Triplet triplet2 =  horizontalSlice.get(i-previousX2,roundedmz);
+				if(triplet2!=null){
+					interpolationY2 = triplet2.intensity;
+					break;
+				}
+			}
+			
+		}
+		
+		double interpolationX =  ((halfHeight - interpolationY1)*((i-previousX2)-i)
+				/(interpolationY2 - interpolationY1))+i;
+		return interpolationX;
+	} 
+		
+	/**
+	 * <p>
+	 * This method is used for getting scan number for given intensity value.
+	 * </p>
+	 * 
+	 * @param height  a {@link java.lang.Double} object. This is intensity value from the horizontal slice from sparse matrix.
+	 */
+	private int getScanNumber(double height){
+		int mu = 0;
+		MapIterator<MultiKey<? extends Integer>, Triplet> iterator = horizontalSlice.mapIterator();
+		
+		while (iterator.hasNext()) {
+			iterator.next();
+
+		   MultiKey<Integer> mk = (MultiKey<Integer>) iterator.getKey();
+		    double intensity = iterator.getValue()!= null?
+		    		(iterator.getValue()).intensity:0;
+
+		    if(intensity == height){
+		    	 mu = (int)mk.getKey(0);
+		    	 break;
+		    }
+		}
+		return mu;
+	}	
+	
+	/**
+	 * <p>
+	 * This method is used calculating BiGaussian values for EIC.
+	 * </p>
+	 * @param x  a {@link java.lang.Integer} object. This is scan number.
+	 */
+	public double getBiGaussianValue(int x){
+		
+		double sigma = x >= BiGaussianParams.mu ? BiGaussianParams.sigmaRight : BiGaussianParams.sigmaLeft;
+		double exponentialTerm = Math.exp(-1 * Math.pow(x-BiGaussianParams.mu, 2) / (2 * Math.pow(sigma,2)));
+		return BiGaussianParams.maxHeight  * exponentialTerm;
+		
+	}
```

Comment:
Note that if you make the variables of ```BiGaussianParams``` non-static, one approach to fix this method would be pass an instance of ```BiGaussianParams``` to this method. If you take that approach, this method can become static since it no longer depends on the state of the ```BiGaussian``` instance. One (major) advantage would be that you can directly unit test this function to ensure its correctness.